tdrs-backend/tdpservice/users/models.py

```diff
@@ -38,15 +38,21 @@ class User(AbstractUser):
         app_label="stts", model="region"
     )
 
-    location_id = models.PositiveIntegerField(null=True, blank=True)
-    location_type = models.ForeignKey(
-        ContentType,
-        on_delete=models.CASCADE,
+    stt = models.ForeignKey(
+        STT,
+        on_delete=models.deletion.SET_NULL,
+        related_name='users',
+        null=True,
+        blank=True
+    )
+
+    region = models.ForeignKey(
+        Region,
+        on_delete=models.deletion.SET_NULL,
```

Comment:
What's the reason for setting it to null instead of cascading deletion?

Reply:
When deleting a "state" STT, a cascading delete would delete any "tribe" stts that were associated. I made the decision to set to `NULL` for the `stt` field thinking since it was a relation to the same model, it might save a headache. For Region, I just copied the same pattern. Open to discussing this further